src/meta_utils.py

```diff
@@ -0,0 +1,528 @@
+            ).merge(
+            fdf, left_on = ["plotid","sampleid"], right_on = ["plotid","sampleid"]
+            )
+        
+        # Insert lon and lat columns
+        meta_dataframe["lon"], meta_dataframe["lat"] = lon, lat
+
+        # Create and add meta features
+        meta_dataframe = create_meta_features(meta_dataframe)
+
+        # Rearrange columns
+        rcolumns = [
+            "plotid", "sampleid", "lon", "lat", "set_1_email", "set_2_email", "overridden_email", 
+            "set_1_analysis_duration", "set_2_analysis_duration", "overridden_analysis", "nonoverridden_analysis", 
+            "set_1_label", "set_2_label", "final_label", "overridden_label"
+        ]
+        meta_dataframe = meta_dataframe[rcolumns]
+
+        return meta_dataframe
+
+    # (2) Else `fdf` is None, then crop mapping
+    else:
+        print("\n{:^53}".format("Creating meta dataframe..."))
+
+        columns.append("crop_noncrop")
+        renamed = lambda s : {
+            "crop_noncrop" : f"{s}_label",
+            "email" : f"{s}_email",
+            "analysis_duration" : f"{s}_analysis_duration"
+        }
+        
+        # Subset dataframes by disagreeing points and columns
+        cdf1 = cdf1.loc[disagreements, columns].rename(columns = renamed("set_1"))
+        cdf2 = cdf2.loc[disagreements, columns].rename(columns = renamed("set_2"))
+
+        # Assemble dataframe
+        meta_dataframe = cdf1.merge(
+            cdf2, left_on = ["plotid", "sampleid"], right_on = ["plotid", "sampleid"]
+        )
+
+        # Insert lon and lat columns
+        meta_dataframe["lon"], meta_dataframe["lat"] = lon, lat
+
+        # Convert analysis duration to float
+        tofloat = lambda string : float(string.split(" ")[0])
+        meta_dataframe[["set_1_analysis_duration", "set_2_analysis_duration"]] = meta_dataframe[["set_1_analysis_duration", "set_2_analysis_duration"]].applymap(tofloat)
+
+        # Rearrange columns
+        rcolumns = [
+            "plotid", "sampleid", "lon", "lat", "set_1_email", "set_2_email", 
+            "set_1_analysis_duration", "set_2_analysis_duration", "set_1_label", "set_2_label", 
+        ]
+        meta_dataframe = meta_dataframe[rcolumns]
+
+        return meta_dataframe
+
+
+def create_meta_dataframe(
+        path_fn : Callable[[str], str],
+        cdate : Optional[str] = None,
+        fdate : Optional[str] = None,
+        area_change : bool = False,
+        y1 : Optional[str] = None,
+        y2 : Optional[str] = None
+    ) -> pd.DataFrame :
+    """ Creates meta dataframe.
+
+    Args:
+
+    Returns:
+    
+    """
+    
+    # (1) Crop **area estimation**
+    #     -> Crop **area**
+    #     -> Crop **area change**
+    if (cdate is not None) and (fdate is not None):
+        # (1.1) Load labeling CSVs to dataframes
+        cdf1, cdf2, fdf = load_dataframes(path_fn, cdate, fdate)
+        
+        # (1.2) If **area change** estimate
```

Comment:
I'd call this comment redundant and thereby not necessary (there's a couple instances around here).

Here's a good resource for comments: https://www.todaysoftmag.com/article/1120/clean-code-comments-and-formatting